osu.Game/Graphics/Containers/ScalingContainer.cs

```diff
@@ -144,6 +145,10 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
+            // for mobile platforms, enforcing 2x scale globally fits well rather than using 1024x768 as target size.
+            if (RuntimeInfo.IsMobile)
+                content.TargetDrawSize = targetMode == ScalingMode.Everything ? DrawSize / 2f : DrawSize;
```

Comment:
What happens if `targetMode` changes?

Reply:
`targetMode` remains constant across all containers, `scalingMode` is what the user can change and that doesn't affect this code. Querying the host window size would probably make it more readable.

Reply:
I've moved code around to hopefully make more sense, and I've also removed Android for now as I'm not confident the scale is good at all.

Reply:
> I've also removed Android for now as I'm not confident the scale is good at all.

I don't think what you're proposing could be worse than `master`. If anything, android devices (unnecessarily) run at (unnecessarily) high pixel densities and would require more tweaking in the upwards direction.

Reply:
The part that I'm confident on is whether my changes won't make the game look too big on Android devices, but I'll push it anyway then.